kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/RequestFilterResultBuilder.java

```diff
@@ -44,4 +44,15 @@ public interface RequestFilterResultBuilder extends FilterResultBuilder<RequestH
      */
     CloseOrTerminalStage<RequestFilterResult> shortCircuitResponse(@NonNull ApiMessage message) throws IllegalArgumentException;
 
+    /**
+     * Generate a short-circuit error response towards the client.
+     *
+     * @param header the headers from the request
+     * @param message the api message to generate an error in response too.
+     * @param throwable the exception that triggered the error response. Note Kafka will map the exception to an {@see org.apache.kafka.common.requests.ApiError} using {@see org.apache.kafka.common.protocol.Errors#forException(java.lang.Throwable)} so callers may wish to supply choose their exception to trigger the appropriate error code
+     * @return next stage in the fluent builder API
+     * @throws IllegalArgumentException header or message do not meet criteria described above.
+     */
+    CloseOrTerminalStage<RequestFilterResult> errorResponse(RequestHeaderData header, ApiMessage message, @NonNull Throwable throwable) throws IllegalArgumentException;
```

Comment:
In general we want to minimise an coupling to the Kafka APIs (both our own and also Filter authors'), especially the ones which are not public APIs of the Kafka project (something is considered public API iff it's in the [published javadoc](https://kafka.apache.org/39/javadoc/org/apache/kafka/common/errors/package-summary.html)). So can be avoid mentioning those classes in this javadoc? I think we could point to `ApiException` in the java, and maybe even use that for the parameter type?

Reply:
> In general we want to minimise an coupling to the Kafka APIs (both our own and also Filter authors')

Really? That seems like an odd goal given our sole reason for existance is to be a proxy that actually understands the RPC's and we are based in Java so we can leverage the serialisers/deseralisers. 

> I think we could point to ApiException in the java, and maybe even use that for the parameter type?

I'd gone with `Throwable` as that matched the Kafka signatures, but walking through the Kafka sources it will do what we want if we tighten this signature to APIError it will map directly and thus avoid walking the stack. I'll update and drop the reference to `Errors`

Reply:
Fixed in https://github.com/kroxylicious/kroxylicious/pull/1630/commits/f1939468381da9cb119f5ac9230c411081292981